src/main/kotlin/ecommerce/dto/ProductRequest.kt

```diff
@@ -1,3 +1,24 @@
 package ecommerce.dto
 
-data class ProductRequest(val name: String, val price: Double, val imageUrl: String)
+import jakarta.validation.constraints.NotBlank
+import jakarta.validation.constraints.Pattern
+import jakarta.validation.constraints.Positive
+import jakarta.validation.constraints.Size
+
+data class ProductRequest(
+    @field:NotBlank(message = "Name must not be blank")
+    @field:Size(max = 15, message = "Name must be at most 15 characters long")
+    @field:Pattern(
+        regexp = "^[\\w\\s\\(\\)\\[\\]\\+\\-\\&/_]*\$",
```

Comment:
```suggestion
        regexp = "^[\\w\\s()\\[\\]+\\-&/_]*$",
```

It seems like there are some redundant escapes 👀